### PR TITLE
HYPRE: Support MPItrampoline

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: H/HYPRE/build_tarballs.jl [hunk 1]
```diff
@@ -47,25 +53,46 @@ cmake .. \
...
+    $(MPI.augment)
+    augment_platform!(platform::Platform) = augment_mpi!(platform)
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms(; experimental=true)
 
+# MPItrampoline requires Fortran
+platforms = expand_gfortran_versions(platforms)
```
Comment (giordano) [inline H/HYPRE/build_tarballs.jl:69]: Same comment as https://github.com/JuliaPackaging/Yggdrasil/pull/5140#discussion_r919457600

Comment (eschnett) [reply H/HYPRE/build_tarballs.jl]: I am trying this.

Comment (eschnett) [reply H/HYPRE/build_tarballs.jl]: Leaving out this statement leads to this error:
```
[ Info: /agent/_work/1/s/H/HYPRE/build/x86_64-linux-gnu-mpi+mpitrampoline/MZd0jNTD/x86_64-linux-gnu-libgfortran3-cxx03-mpi+mpitrampoline/destdir/lib/libHYPRE.so cannot be dlopen'ed
```

See e.g. https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=20647&view=logs&j=7aeb1989-ab2c-51b9-49d6-09383b673c9a&t=f7086e79-1917-5649-bdc5-1508d4a22ae4

Comment (eschnett) [reply H/HYPRE/build_tarballs.jl]: Please advise how to proceed. Should we debug this, or just put the expansion back in?

Comment (giordano) [reply H/HYPRE/build_tarballs.jl]: Isn't this _precisely_ the same situation as https://github.com/JuliaPackaging/Yggdrasil/pull/5028#issuecomment-1165869962?

Comment (eschnett) [reply H/HYPRE/build_tarballs.jl]: Yes!

**Pull request merged**